### PR TITLE
initial bart implementation

### PR DESCRIPTION
**Pull request by Christof23**

### CODE REVIEW

**Comment by LaurentMazare** [comment]
Looks like a great model to have, thanks for porting this. Before reviewing the code could you:
- Remove the commented code parts.
- Remove the `allow` pragmas.
```
#![allow(unused_imports)]
#![allow(dead_code)]
#![allow(unused_variables)]
#![allow(unused_mut)]
```
- Detail what testing you've done, e.g. comparing the resulting embeddings with the python version etc.
- Provide how to run the example on some sample sentence in the readme?

**Comment by LaurentMazare** [comment]
Looks like you've been pushing on these and on the shape mismatch, that's great! Let me know when I can have a look again!

**Comment by Christof23** [comment]
@LaurentMazare I'm going to run checks on the tensor output now that the full forward pass for sequence classification has been implemented. Will let you know once this is done!

**Comment by LaurentMazare** [comment]
Great, looks super exciting to have this model family! Just eyeballing the diff, you may want to replace a bunch of `unwrap()` with `?` so as to enable proper error reporting rather than panicking. Thanks!

**Comment by Christof23** [comment]
Looks like `rust-bert` doesn't follow the transformers implementation for BART super precisely, will take a little while longer to get a mirrored version...

**Comment by LaurentMazare** [comment]
Gotcha, I have to say that so far I've ported models directly from the python transformers implementation, it's probably a bit more conversion work to get it to compile but indeed likely easier to line up afterwards.

**Comment by LaurentMazare** [comment]
Any update on this? This would make for a great addition once the logits have been lined up with a reference implementation.

**Comment by LaurentMazare** [comment]
Closing this as now recent update.
